src/fluree/server/handler.clj

```diff
@@ -231,6 +232,39 @@
           fuel 1000] ; TODO: get this for real
       (assoc-in resp [:headers "x-fdb-fuel"] (str fuel)))))
 
+(defn wrap-policy-metadata
+  "Both Fluree-Policy-Class and Fluree-Policy-Identity can be optionally
+  set in the header for non-credentialed queries/transactions. The primary
+  motivation is enforcing policy with SPARQL which doesn't have an opts
+  map option, although if set, it will also override policyClass/Identity
+  that might have otherwise been set in FlureeQL/JSON."
+  [handler]
+  (fn [req]
+    (let [policy-identity (get-in req [:headers "fluree-policy-identity"])
+          policy-class    (get-in req [:headers "fluree-policy-class"])
+          policy          (when-let [p (get-in req [:headers "fluree-policy"])]
+                            (try
+                              (json/parse p false)
+                              (catch Exception _
+                                (throw (ex-info "Invalid policy header, it must be parsable JSON."
```

Comment:
A more precise error message might be more useful: "Invalid fluree-policy-class header: must be JSON".

Reply:
Done!